core/src/main/scala/org/http4s/otel4s/middleware/ServerMiddleware.scala

```diff
@@ -150,26 +150,24 @@ object ServerMiddleware {
                 .build
                 .use { span =>
                   poll(f.run(req))
-                    .guaranteeCase {
-                      case Outcome.Succeeded(fa) =>
-                        span.addAttribute(Attribute("exit.case", "succeeded")) >>
+                    .guaranteeCase { outcome =>
```

Comment:
Curious: did a linter or compiler complain about the partial function being the function literal here, or is this a stylistic choice?  I don't mind either way, but it's the part of this that puzzled me most.